### PR TITLE
libpod: unlock containers when removing pod

### PR DESCRIPTION
**Pull request by giuseppe**
It solves a race where a container cleanup process launched because of
the container process exiting normally would hang.

It also solves a problem when running as rootless on cgroup v1 since
it is not possible to force pids.max = 1 on conmon to limit spawning
the cleanup process.

Partially copied from https://github.com/containers/podman/pull/13403

Related to: https://github.com/containers/podman/issues/14057

[NO NEW TESTS NEEDED] it doesn't add any new functionality

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM

File: libpod/runtime_pod_linux.go [hunk 1]
```diff
@@ -304,6 +309,12 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool,
 		}
 	}
 
+	// let's unlock the containers so if there is any cleanup process, it can terminate its execution
+	for _, ctr := range ctrs {
+		ctr.lock.Unlock()
```
Comment (flouthoc) [inline libpod/runtime_pod_linux.go:314]: Just a small non-blocker concern here: Would this allow other podman process to modify ctrs while the pod is being removed same as https://github.com/containers/podman/pull/13403#discussion_r817596145

**Review by flouthoc — APPROVED** [review]
LGTM just one non-blocking question above

File: libpod/runtime_pod_linux.go (hunk 1 above)
Comment (flouthoc) [reply libpod/runtime_pod_linux.go:314]: nvm we are already removing all containers from state before this so I think this should be only useful for another cleanup process.

**Pull request merged**